dspy/clients/lm.py

```diff
@@ -212,7 +216,7 @@ def copy(self, **kwargs):
         return new_instance
 
 
-@functools.lru_cache(maxsize=None)
+# @functools.lru_cache(maxsize=None)
```

Comment:
This is a hack to get the implementation working end to end. We need a proper fix, e.g. https://github.com/stanfordnlp/dspy/pull/1862 (though it's not 100% clear to me why we need LRU caching here in the first place on top of the caching that LiteLLM is already providing)

Reply:
cc @okhat I'm sure I'm missing something here - let me know if there's additional context